tests/integration/conftest.py

```diff
@@ -58,3 +72,34 @@ def bans_url_1(configured_httpserver):
 @pytest.fixture
 def bans_url_2(configured_httpserver):
     return configured_httpserver.url_for("/bansummary-changed.text")
+
+
+@pytest.fixture
+def sequential_handler(request):
+    if hasattr(request, "param"):
+        responses = request.param
+    else:
+        responses = []
+        # Any test that requests this fixture should perform indirect parameterization to set the sequence of expected responses.
+
+    # Create a generator out of sequence of respnoses
+    resp_generator = (v for v in responses)
+
+    # Handler returns next item in generator, and state is maintained until fixture is destroyed
+    def my_handler(req):
+        return next(resp_generator)
+
+    return my_handler
+
+
+@pytest.fixture
+def sequential_httpserver(httpserver, sequential_handler):
+    httpserver.expect_request(
+        "/dynamic-dynamicconfig.ini", method="GET"
+    ).respond_with_handler(sequential_handler)
+    return httpserver
```

Comment:
This httpserver will respond with the above handler whenever this custom url is hit. Therefore, the webserver will respond with potentially different content upon multiple requests.